synapse/cortex.py

```diff
@@ -2139,47 +2139,81 @@ async def _tryLoadStormPkg(self, pkgdef):
             name = pkgdef.get('name', '')
             logger.exception(f'Error loading pkg: {name}, {str(e)}')
 
-    async def _reqStormPkgDeps(self, pkgdef):
+    async def verifyStormPkgDeps(self, pkgdef):
+
+        result = {
+            'requires': [],
+            'conflicts': [],
+        }
 
         deps = pkgdef.get('depends')
         if deps is None:
-            return
-
-        name = pkgdef.get('name')
+            return result
 
         requires = deps.get('requires', ())
         for require in requires:
 
             pkgname = require.get('name')
             cmprvers = require.get('version')
 
+            item = require.copy()
+            item.setdefault('desc', None)
+
             cpkg = await self.getStormPkg(pkgname)
-            if cpkg is not None:
+
+            if cpkg is None:
+                item.update({'ok': False, 'actual': None})
+            else:
                 cver = cpkg.get('version')
-                if s_version.matches(cver, cmprvers):
-                    continue
+                ok = s_version.matches(cver, cmprvers)
+                item.update({'ok': ok, 'actual': cver})
 
-            mesg = f'Storm package {name} requires {pkgname}{cmprvers}.'
-            raise s_exc.StormPkgRequires(mesg=mesg)
+            result['requires'].append(item)
 
         conflicts = deps.get('conflicts', ())
         for conflict in conflicts:
 
             pkgname = conflict.get('name')
+            cmprvers = conflict.get('version')
+
+            item = conflict.copy()
+            item.setdefault('version', None)
+            item.setdefault('desc', None)
 
             cpkg = await self.getStormPkg(pkgname)
+
             if cpkg is None:
+                item.update({'ok': True, 'actual': None})
+            else:
+                cver = cpkg.get('version')
+                ok = cmprvers is not None and not s_version.matches(cver, cmprvers)
+                item.update({'ok': ok, 'actual': cver})
+
+            result['conflicts'].append(item)
+
+        return result
+
+    async def _reqStormPkgDeps(self, pkgdef):
+
+        name = pkgdef.get('name')
+
+        deps = await self.verifyStormPkgDeps(pkgdef)
+
+        for require in deps['requires']:
+
+            if require['ok']:
                 continue
 
-            cmprvers = conflict.get('version')
-            if cmprvers is None:
-                mesg = f'Storm package {name} conflicts with {pkgname}.'
-                raise s_exc.StormPkgConflicts(mesg=mesg)
-
-            cver = cpkg.get('version')
-            if s_version.matches(cver, cmprvers):
-                mesg = f'Storm package {name} conflicts with {pkgname}{cmprvers}.'
-                raise s_exc.StormPkgConflicts(mesg=mesg)
+            mesg = f'Storm package {name} requires {require.get("name")}{require.get("version")}.'
+            logger.warning(mesg)
```

Comment:
Is this going to happen a lot when things load in an arbitrary order? Just looking to avoid "WARNING: everything is fine!"

Reply:
Updated to be INFO log with lang "...not currently met..."